packages/frontend/src/providers/DashboardProvider.tsx

```diff
@@ -133,22 +202,44 @@ export const DashboardProvider: React.FC = ({ children }) => {
         return uniqBy(allFilters, (f) => fieldId(f));
     }, [isLoadingDashboardFilters, filterableFieldsBySavedQueryUuid]);
 
-    const [fieldsWithSuggestions, setFieldsWithSuggestions] =
-        useState<FieldsWithSuggestions>({});
-    const [dashboardTemporaryFilters, setDashboardTemporaryFilters] =
-        useState<DashboardFilters>(emptyFilters);
-    const [dashboardFilters, setDashboardFilters] =
-        useState<DashboardFilters>(emptyFilters);
-    const [haveFiltersChanged, setHaveFiltersChanged] =
-        useState<boolean>(false);
+    const allFilters = useMemo(() => {
+        return {
+            dimensions: [
+                ...dashboardFilters.dimensions,
+                ...dashboardTemporaryFilters?.dimensions,
+            ],
+            metrics: [
+                ...dashboardFilters.metrics,
+                ...dashboardTemporaryFilters?.metrics,
+            ],
+        };
+    }, [dashboardFilters, dashboardTemporaryFilters]);
 
-    // Set filters to filters from database
-    useEffect(() => {
-        if (dashboard && dashboardFilters === emptyFilters) {
-            setDashboardFilters(dashboard.filters);
-            setHaveFiltersChanged(false);
+    const hasChartTiles = useMemo(
+        () =>
+            dashboardTiles.filter(
+                (tile) => tile.type === DashboardTileTypes.SAVED_CHART,
+            ).length >= 1,
+        [dashboardTiles],
+    );
+
+    useMemo(() => {
```

Comment:
I don't understand the change from useEffect to useMemo. useEffect seems to be the correct since the hook doesn't return a value.

Reply:
It shouldn't make much difference, I tend to follow the react docs best practices and use useEffect for side effects like fetches and the like. And useMemo for caching computed values, like we're doing here. 

Too many useEffects on a page tends to make it really hard to follow the logic, but I don't have a strong opinion. I'm happy to change this back if you prefer. Or if I am missing something about how this would break 😄

Reply:
> And useMemo for caching computed values, like we're doing here.

But we are not caching/memoizing the computed value.

When I'm scanning the code and see a useMemo I know all the logic for that variable is encapsulated in there. In this case the value is stored in a different state `const [fieldsWithSuggestions, setFieldsWithSuggestions] = useState<FieldsWithSuggestions>({});` that is also being updatedin other places.

> I tend to follow the react docs best practices and use useEffect for side effects like fetches and the like.

Can you find examples where useMemo doesn't return a value?
Because the golden rule I've followed: 
- if it returns a value -> useMemo
- if it returns a function -> useCallback
- if doesn't return anything -> useEffect

Reply:
That's fine with me. This can be a useEffect.